packages/common/src/failure.ts

```diff
@@ -329,15 +329,22 @@ export async function errorToFailure(err: unknown, dataConverter: DataConverter)
     source: FAILURE_SOURCE,
   };
 
-  if (err instanceof Error) {
-    return { ...base, message: err.message ?? '', stackTrace: cutoffStackTrace(err.stack) };
+  if (isRecord(err) && hasOwnProperties(err, ['message', 'stack'])) {
+    return {
+      ...base,
+      message: String(err.message) ?? '',
+      stackTrace: cutoffStackTrace(String(err.stack)),
+      cause: await optionalErrorToOptionalFailure(err.cause, dataConverter),
```

Comment:
Until this change `cause` was only used for chaining TemporalFailures, I don't know if we chain error causes in the other SDKs and if we should.

Reply:
I haven't found where in Go SDK workflow errors are handled, but in Java, looks like `Error`s are rethrown and other non-specific throwables are wrapped in `WorkflowServiceException` with the throwable as the `cause`:

https://github.com/temporalio/sdk-java/blob/b0199150b69bb5a4eaf6c13ba4b89c2a8746a113/temporal-sdk/src/main/java/io/temporal/internal/sync/WorkflowStubImpl.java#L299-L326

I'm not sure whether uniformity is important here? I figure we have a `cause` field, and might as well use it, as it could be helpful for users who are catching and throwing new non-`ApplicationFailure`s and want to include the original error as a cause. It will probably become a more common pattern (was added to v8 last year).

Reply:
It will become popular I suspect, now it requires node > 16.9 AFAIK.

Let's keep it, sounds like a good addition to me.